src/Test/AbstractIntegrationTestCase.php

```diff
@@ -243,39 +245,41 @@ protected function doTest(IntegrationCase $case)
-                    return $fixer->getPriority();
-                },
-                $fixers
-            );
-
-            $this->assertNotCount(1, array_unique($priorities), sprintf('All used fixers must not have the same priority, integration tests should cover fixers with different priorities. In "%s".', $case->getFileName()));
-
-            $tmpFile = static::getTempFile();
-            if (false === @file_put_contents($tmpFile, $input)) {
-                throw new IOException(sprintf('Failed to write to tmp. file "%s".', $tmpFile));
-            }
-
-            $runner = new Runner(
-                new \ArrayIterator(array(new \SplFileInfo($tmpFile))),
-                array_reverse($fixers),
-                new SebastianBergmannDiffer(),
-                null,
-                $errorsManager,
-                $this->linter,
-                false,
-                new NullCacheManager()
-            );
+        $tmpFile = static::getTempFile();
+        if (false === @file_put_contents($tmpFile, $input)) {
+            throw new IOException(sprintf('Failed to write to tmp. file "%s".', $tmpFile));
+        }
 
-            $runner->fix();
-            $fixedInputCodeWithReversedFixers = file_get_contents($tmpFile);
+        $runner = new Runner(
+            new \ArrayIterator(array(new \SplFileInfo($tmpFile))),
+            array_reverse($fixers),
+            new SebastianBergmannDiffer(),
+            null,
+            $errorsManager,
+            $this->linter,
+            false,
+            new NullCacheManager()
+        );
 
-            $this->assertNotSame(
-                $fixedInputCode,
-                $fixedInputCodeWithReversedFixers,
-                sprintf('Set priorities must be significant. If fixers used in reverse order return same output then the integration test is not sufficient or the priority relation between used fixers should not be set. In "%s".', $case->getFileName())
+        Tokens::clearCache();
+        $runner->fix();
+        $fixedInputCodeWithReversedFixers = file_get_contents($tmpFile);
+
+        // If output is different depends on rules order - we need to verify that the rules are ordered by priority.
+        // If not, any order is valid.
+        if ($fixedInputCode !== $fixedInputCodeWithReversedFixers) {
+            $this->assertGreaterThan(
+                1,
+                count(array_unique(array_map(
+                    function (FixerInterface $fixer) {
+                        return $fixer->getPriority();
+                    },
+                    $fixers
+                ))),
+                sprintf(
+                    'Rules priorities are not differential enough. If rules would be used in reverse order then final output would be different than the expected one. For that, different priorities must be set up for used rules to ensure stable order of them. In "%s".',
+                    $case->getFileName()
+                )
```

Comment:
> Maybe I miss read the PR than, as now the AbstractIntegrationTestCase will always test in reverse order now so the set tests are effected as well aren't they?

master version:
- test reverse order only when `checkPriority` is true (in my PR - always)
- `set` may or may not be tested in revers order, i don't know, i don't care - they are outside of `priority` subdir

> Secondly since there is no assert for when the content is the same in reverse order we loose the detection of invalid order dependency claims 

I add it at first, and then realize there are some problems where A and B fixers could run in any order, but have different priorities because of relations A<->C and B<->D